processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -341,7 +435,18 @@ public void advance()
                       if (Thread.interrupted()) {
                         throw new QueryInterruptedException(new InterruptedException());
                       }
-                      cursorOffset.increment();
+                      cursorOffsetHolder.get().increment();
+
+                      while (!isDone()) {
```

Comment:
Is this the case even for really simple queries? Like a timeseries with no post-filters a single longSum. Usually if there are any inefficiences in the cursor they will show up in a query like that.